controller_interface/include/controller_interface/controller_interface.hpp

```diff
@@ -20,7 +20,7 @@
 
 #include "controller_interface/visibility_control.h"
 
-#include "hardware_interface/robot_hardware.hpp"
+#include "hardware_interface/resource_manager.hpp"
```

Comment:
I don't think that's the right approach. The hardware interface should stay agnostic of the resource manager in my opinion. I consider the resource manager as an internal detail of the controller manager.
The reason I am seeing here is that the hardware interfaces could technically be used outside the controller manager's resource handling.

Reply:
In general, I agree with you, only issue here are potentially cyclic dependencies if the `resource_manager` is in the `controller_manger` package: 

controller_manager -> controller_interface -> resource_manager

A solution could be to move `resource_manager` into its own package.

Reply:
see for more discussion: https://github.com/ros-controls/ros2_control/issues/174

Reply:
I'd strongly advise against creating new packages to resolve internal dependency issues.